### PR TITLE
Improved memory space display

### PR DESCRIPTION
**Pull request by bobjacobsen**
- Add valid/present flag to Address Space reply in the memory configuration service
 - Display additional information in the MemConfigDescriptionPane

### CODE REVIEW

**Comment by bobjacobsen** [comment]
Also includes 
 - Fix to Hub to handle timeout seen (only) on Windows machines, despite Javadoc saying that socket read timeout==0 should prevent that.
 - Minor improvement to how e.g JMRI nodes are shown in config dialog
 - Improve how timers are handled in multiple config dialog windows

File: src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java [hunk 1]
```diff
@@ -81,10 +85,14 @@ public void handleFailure(int code) {
...
                     // fill window from values
                     commandLabel.setText("0x"+Utilities.toHexPair(commands>>8)+Utilities.toHexPair(commands));
+                    writesLabel.setText("0x"+Utilities.toHexPair(lengths));
 
                     highSpaceLabel.setText("0x"+Utilities.toHexPair(highSpace));
                     lowSpaceLabel.setText("0x"+Utilities.toHexPair(lowSpace));
 
+                    // sometimes highSpace doesn't include the pre-defined spaces,
+                    // so in the case we add them on and start with 0xFF
+                    if (highSpace >= 0xFA) highSpace = 0xFF;
```
Comment (balazsracz) [inline src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java:95]: isn't this intended to be < 0xFA ?

File: src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java [hunk 2]
```diff
@@ -50,6 +53,7 @@ public MemConfigDescriptionPane(NodeID node, MimicNodeStore store,
         setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
 
         addLine(commandLabel, "Commands:");
+        addLine(writesLabel, "Write Codes:");
```
Comment (balazsracz) [inline src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java:56]: How about "Writes Supported:" or "Write Modes:"

**Review by balazsracz — APPROVED** [review]

File: src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java (hunk 2 above)
Comment (bobjacobsen) [reply src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java]: Change to Write Modes, thanks for suggestion.

File: src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java (hunk 1 above)
Comment (bobjacobsen) [reply src/org/openlcb/swing/memconfig/MemConfigDescriptionPane.java:95]: Certain nodes show 0xFA even though they have spaces up to 0xFF.  So this is meant as a Gentle Hack to include those upper spaces in the display.  Hence the = part of >= 0xFA.  The > part is an attempt at generality if some future node claims a top of 0xFB, etc.

**Pull request merged**